plugins/modules/k8s_exec.py

```diff
@@ -131,15 +165,26 @@ def main():
         stdin=False,
         tty=False,
         _preload_content=False, **optional_kwargs)
-    stdout, stderr = [], []
+    stdout, stderr, rc = [], [], 0
     while resp.is_open():
         resp.update(timeout=1)
         if resp.peek_stdout():
             stdout.append(resp.read_stdout())
         if resp.peek_stderr():
             stderr.append(resp.read_stderr())
+    err = resp.read_channel(3)
+    err = yaml.safe_load(err)
+    if err['status'] == 'Success':
+        rc = 0
+    else:
+        rc = int(err['details']['causes'][0]['message'])
+    changed = True if rc == 0 else False
```

Comment:
Does it make sense to assume no change on failure? Many changes could still be made by a complex command that ultimately failed

Reply:
That's what I was thinking. I will keep the original condition of `changed` (always returning True). Thanks for confirming.